underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -83,6 +93,9 @@ public int read(byte[] b, int offset, int length) throws IOException {
     int read = mStream.read(b, offset, length);
     if (read != -1) {
       mPos += read;
+      if (mPos % getBlockSize() == 0) {
+        closeStream();
+      }
```

Comment:
Yeah for scenario 1, but I think that is okay because we will return `10` since we didn't break any contracts. The caller would need to call read in a loop to get the full data.

You are right for scenario 2, but then when will we ever return `-1`? Will `openStream` do the right thing when the range is beyond the file size? Or will we get an exception from the Swift library?